Automattic/wp-calypso
# Gravatar: Implement passwordless login flow

Related to 105630-gh-Automattic/gravatar

## Proposed Changes

* Customize the new Gravatar login page
* Customize the new Gravatar magic login page and enable new account creation
* Customize the new Gravatar email verification page and implement the resend email feature
* Clear the old Gravatar login page-related code
* Fix the wrong signup URLs on the related F2A pages
* Fix an existing lint error by using `localizeUrl` for the "Get help" link for F2A pages

## Testing Instructions



1. Setup Calypso repo by following [this doc](https://github.com/Automattic/wp-calypso#getting-started) and check out this PR
2. Run `yarn install` and then `yarn start`
3. Logout your account, and open [this magic login page](http://calypso.localhost:3000/log-in/link?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D111ffb2efc6d68ee7a6379a556cc6a9195c2ad255a2f310f7042213fd5c4ee29%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854)
4. Try to **signup a new account** (you can use [Gmail testing accounts](https://www.google.com/search?q=gmail+test+accounts&rlz=1C5CHFA_enTW1017TW1017&oq=gmail+test+accounts&gs_lcrp=EgZjaHJvbWUqBggAEEUYOzIGCAAQRRg7MggIARAAGBYYHjIICAIQABgWGB4yCggDEAAYhgMYigUyCggEEAAYhgMYigUyCggFEAAYhgMYigUyBggGEEUYPDIGCAcQRRg80gEINTAzNWowajeoAgCwAgA&sourceid=chrome&ie=UTF-8)), and the flow should work as below:

https://github.com/Automattic/wp-calypso/assets/21308003/668349e8-027a-42e7-9af5-ab3d23fecf30

5.  Logout your account, and try to **login with a non-A11n + F2A account** via [this magic login page](http://calypso.localhost:3000/log-in/link?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D111ffb2efc6d68ee7a6379a556cc6a9195c2ad255a2f310f7042213fd5c4ee29%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854), the flow should work as below:

https://github.com/Automattic/wp-calypso/assets/21308003/d9d3e2b4-f63e-4f23-b209-222c1031c360

6. Check the UIs and click the following links, they should work as expected:

| Magic login | Check your email |
| - | - |
| <img width="665" alt="截圖 2023-09-01 下午3 11 08" src="https://github.com/Automattic/wp-calypso/assets/21308003/e302c599-6da6-4426-906f-ff134ecc836d"> | <img width="642" alt="截圖 2023-09-01 下午3 11 29" src="https://github.com/Automattic/wp-calypso/assets/21308003/5c84a771-354a-4db8-9597-296478308aff"> |

7.  Logout your account, and try to **login with a passwordless account (like we did via No.5)** via [this login page](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D111ffb2efc6d68ee7a6379a556cc6a9195c2ad255a2f310f7042213fd5c4ee29%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854), and the flow should work as below:

https://github.com/Automattic/wp-calypso/assets/21308003/840802f4-616b-4a6f-9e17-ba62cfe6e9af

8. Check the UIs and click the following links, they should work as expected:

<img width="582" alt="截圖 2023-09-01 下午3 46 49" src="https://github.com/Automattic/wp-calypso/assets/21308003/07bacc6f-2688-4b73-9dba-4d2adda9ef54">

9. Login with an F2A account via [this login page](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D111ffb2efc6d68ee7a6379a556cc6a9195c2ad255a2f310f7042213fd5c4ee29%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854), and click the following signup links, you will be redirect to Gravatar's signup page:

| F2A app page | F2A code page | F2A backup code page |
| - | - | - |
| <img width="833" alt="截圖 2023-09-01 下午4 12 00" src="https://github.com/Automattic/wp-calypso/assets/21308003/726b94d2-fc6a-4b51-8dbc-49886b67b241"> |  <img width="834" alt="截圖 2023-09-01 下午4 12 07" src="https://github.com/Automattic/wp-calypso/assets/21308003/ef8aab81-bd0e-45e6-8369-b595797dca5e"> | <img width="834" alt="截圖 2023-09-01 下午4 12 15" src="https://github.com/Automattic/wp-calypso/assets/21308003/fde1c43c-d9ef-4208-abd9-279f480fc813"> |

10. Check WPCOM's login page, it should work the same:
  - [Dev](http://calypso.localhost:3000/log-in)
  - [Prod](https://wordpress.com/log-in)

11. Check WPCOM's magic login page, it should work the same:
  - [Dev](http://calypso.localhost:3000/log-in/link)
  - [Prod](https://wordpress.com/log-in/link)

12. Check any OAuth2 clients, e.g. Jetpack, it should work the same:
  - [Dev](http://calypso.localhost:3000/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)
  - [Prod](https://wordpress.com/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)

## Pre-merge Checklist
